### ISSUE TITLE
ssh protocol support on mac

### ISSUE THREAD

**Issue by hadley** [ropensci/git2r#267]
The CRAN version of git2r doesn't seem to support the ssh protocol. Any idea what's wrong?

**Comment by stewid** [comment]
@hadley thanks for reporting the problem. I'll debug it tomorrow.

**Comment by stewid** [comment]
I have not been able to find the problem. What's the output from `libgit2_features` (I don't have mac)?

**Comment by hadley** [comment]
```
> git2r::libgit2_features()
$threads
[1] FALSE

$https
[1] TRUE

$ssh
[1] FALSE
```

**Comment by stewid** [comment]
Thanks. I have to improve the configuration and download the latest release of libssh2 from https://github.com/libssh2/libssh2 and build it locally if homebrew isn't available. I fix that this week.

**Comment by stewid** [comment]
@hadley could you please test ssh transport on mac with the new CRAN release (v0.17.0).

Thanks

**Comment by hadley** [comment]
Doesn't look good 😦 

```
>  git2r::libgit2_features()
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/Users/hadley/R/git2r/libs/git2r.so':
  dlopen(/Users/hadley/R/git2r/libs/git2r.so, 6): Symbol not found: _EVP_aes_128_cbc
  Referenced from: /Users/hadley/R/git2r/libs/git2r.so
  Expected in: flat namespace
 in /Users/hadley/R/git2r/libs/git2r.so
```

**Comment by jimhester** [comment]
I get the same thing as hadley FWIW

**Comment by nathaneastwood** [comment]
I'm also have issues, primarily with `devtools::release()`. But this is my current output

```
> git2r::libgit2_features()
Error in get(method, envir = home) : 
  lazy-load database '/Library/Frameworks/R.framework/Versions/3.3/Resources/library/git2r/R/git2r.rdb' is corrupt
In addition: Warning messages:
1: In .registerS3method(fin[i, 1], fin[i, 2], fin[i, 3], fin[i, 4],  :
  restarting interrupted promise evaluation
2: In get(method, envir = home) :
  restarting interrupted promise evaluation
3: In get(method, envir = home) : internal error -3 in R_decompress1
```

This is a brand new MBPr with a fresh install of homebrew, R, RStudio and `git2r` and `devtools`.

**Comment by hadley** [comment]
@nathaneastwood that's a different problem and is usually fixed by restarting R

**Comment by nathaneastwood** [comment]
Hmm, fair enough. I tried restarting R but no joy ☹️

**Comment by stewid** [comment]
No, that doesn't look good. I have to investigate this further, but first I submit a new relase tomorrow that doesn't download and build libssh2 on mac.

**Comment by stewid** [comment]
I found an error in the configure script that I think caused the linking problem. A new version (0.18.0) is on CRAN, could you please test if SSH transport works on mac now.

**Comment by hadley** [comment]
Looks good!

```R
> git2r::libgit2_features()
$threads
[1] FALSE

$https
[1] TRUE

$ssh
[1] TRUE
```

**Comment by LechMadeyski** [comment]
Neither CRAN version of `git2r` nor dev version (installed via `install_github("ropensci/git2r")`) seems to support the ssh protocol. `libssh2` installed via `brew install libssh2`.

Any idea what is wrong? 
  
```libgit2_features()
$threads
[1] FALSE

$https
[1] TRUE

$ssh
[1] FALSE

**Comment by stewid** [comment]
@LechMadeyski does it work if you install git2r with the following command?

```
install.packages("git2r", type="source", configure.vars="autobrew=yes")
```

**Comment by LechMadeyski** [comment]
Unfortunately not. I installed and tested, removed git2r and installed once again. Nothing has changed.

```
> git2r::libgit2_features()
$threads
[1] FALSE

$https
[1] TRUE

$ssh
[1] FALSE
```

**Comment by antass** [comment]
Had the same issue (no SSH support despite installed `libssh2` and `libgit2` on MacOS), but @stewid's suggestion (installing from CRAN from source with `autobrew` enabled) worked! Thanks!

**Comment by jzadra** [comment]
> install.packages("git2r", type="source", configure.vars="autobrew=yes")

This fixed the issue for me.

**Comment by AZhyrenkov** [comment]
For me it doesn't help, unfortunately:

```remove.packages('git2r')
install.packages("git2r", type="source", configure.vars="autobrew=yes")

library(git2r)
library(devtools)
libgit2_features()
```

shows 
```
$threads
[1] FALSE

$https
[1] TRUE

$ssh
[1] FALSE
```

libssh2 is installed:
```
brew reinstall libssh2
==> Downloading https://homebrew.bintray.com/bottles/libssh2-1.9.0_1.catalina.bottle.tar.gz
Already downloaded: /--users-dir--/Library/Caches/Homebrew/downloads/73b0bbe5fc3658c307a3f43903218f000f99a4d5cdc6cd23f67e88eea39d9e49--libssh2-1.9.0_1.catalina.bottle.tar.gz
==> Reinstalling libssh2
==> Pouring libssh2-1.9.0_1.catalina.bottle.tar.gz
```

```
> session_info()
─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       macOS Catalina 10.15.4      
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Europe/Kiev                 
 date     2020-05-27                  

─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version date       lib source        
 assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.6.3)
 backports     1.1.5   2019-10-02 [1] CRAN (R 3.6.3)
 callr         3.4.3   2020-03-28 [1] CRAN (R 3.6.2)
 cli           2.0.2   2020-02-28 [1] CRAN (R 3.6.3)
 crayon        1.3.4   2017-09-16 [1] CRAN (R 3.6.3)
 desc          1.2.0   2018-05-01 [1] CRAN (R 3.6.3)
 devtools    * 2.3.0   2020-04-10 [1] CRAN (R 3.6.3)
 digest        0.6.25  2020-02-23 [1] CRAN (R 3.6.3)
 ellipsis      0.3.0   2019-09-20 [1] CRAN (R 3.6.3)
 fansi         0.4.1   2020-01-08 [1] CRAN (R 3.6.3)
 fs            1.3.2   2020-03-05 [1] CRAN (R 3.6.0)
 git2r       * 0.27.1  2020-05-03 [1] CRAN (R 3.6.3)
 glue          1.4.1   2020-05-13 [1] CRAN (R 3.6.2)
 magrittr      1.5     2014-11-22 [1] CRAN (R 3.6.3)
 memoise       1.1.0   2017-04-21 [1] CRAN (R 3.6.0)
 pkgbuild      1.0.6   2019-10-09 [1] CRAN (R 3.6.3)
 pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.6.3)
 prettyunits   1.1.1   2020-01-24 [1] CRAN (R 3.6.3)
 processx      3.4.2   2020-02-09 [1] CRAN (R 3.6.3)
 ps            1.3.2   2020-02-13 [1] CRAN (R 3.6.3)
 R6            2.4.1   2019-11-12 [1] CRAN (R 3.6.3)
 Rcpp          1.0.4.6 2020-04-09 [1] CRAN (R 3.6.3)
 remotes       2.1.1   2020-02-15 [1] CRAN (R 3.6.0)
 rlang         0.4.6   2020-05-02 [1] CRAN (R 3.6.2)
 rprojroot     1.3-2   2018-01-03 [1] CRAN (R 3.6.3)
 rstudioapi    0.11    2020-02-07 [1] CRAN (R 3.6.3)
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.6.0)
 testthat      2.3.2   2020-03-02 [1] CRAN (R 3.6.3)
 usethis     * 1.6.0   2020-04-09 [1] CRAN (R 3.6.3)
 withr         2.1.2   2018-03-15 [1] CRAN (R 3.6.3)

[1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

**Comment by ordiology** [comment]
I know this issue is closed, but I had the same problem and none of the above worked. I thought I'd share my fix and save some frustration for others. I found I was missing libgit2.

This worked for me...
brew install libssh2
brew install libgit2.

After that, re-install git2r from source with...
install.packages("git2r", type="source", configure.vars="autobrew=yes")

I found the solution here: 
https://stackoverflow.com/questions/60176959/getting-unsupported-url-protocol-when-using-ssh-authentication-with-git2rclo
